testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6421,7 +6421,11 @@ private static void checkIf(SqlOperatorFixture f) {
       f.checkType("atanh('abc')", "DOUBLE NOT NULL");
       f.checkScalarApprox("atanh(0.76159416)", "DOUBLE NOT NULL",
           isWithin(1d, 0.0001d));
+      f.checkScalarApprox("atanh(cast(-0.1 as decimal(19, 0)))", "DOUBLE NOT NULL",
```

Comment:
Do we need `atanh` and `ceil` to hit the problem or it is sufficient to use `CAST`? If the problem is reproable simply with `CAST` then we should simply add test cases for that.

Reply:
Just compare with the line 6425.